deeppavlov/core/commands/train.py

```diff
@@ -92,6 +92,11 @@ def train_evaluate_model_from_config(config: Union[str, Path, dict],
     if iterator is None:
         try:
             data = read_data_by_config(config)
+            # if the value 'val_every_n_epochs' is set in the config and there is no validation data - raise exception
+            if config.get('train', {}).get('val_every_n_epochs', None) and not data.get('valid', []):
```

Comment:
add batch parameter check, replace `not data.get('valid', [])` with `not data.get('valid')`